### PR TITLE
initial demo

### PR DESCRIPTION
**Pull request by flora8984461**
New todo - Updated on 2020/12/04:

- [x] Be able to add / delete fields of representatives
- [x] Make participation level to be dependent on working groups
- [x] Better validation
- [ ] My own todo: need to restructure the code to make it cleaner and better maintain

-------------------

1. Be able to add / delete fields of respresentative
2. Company dropdown to prefill the company info
3. What is your intended Membership Level? Add MEMBERSHIP FEES table
4. Add correct membership levels and working groups
5. participation level can be the same step as working groups, it depends on working groups + Effective Date + Working Group Representative
6. ...

-------------------
Current version:
According to https://docs.google.com/document/d/1E1THXZ-YhL_Rp5wMKD79O9Hb1ynQsrlqagnNJE3EPr8/edit# ,

Notes:

Links should be:  
Go to login firstly, choose your membership status, and the form will handle the status
https://deploy-preview-1--eclipsefdn-react-members.netlify.app/login 
https://deploy-preview-1--eclipsefdn-react-members.netlify.app/form

Signed-off-by: Yi Liu <yi.liu@eclipse-foundation.org>

### CODE REVIEW

**Comment by autumnfound** [comment]
@flora8984461 would you like review on this now, or are you still working on it?

**Comment by flora8984461** [comment]
> @flora8984461 would you like review on this now, or are you still working on it?

It would be great if you have some thoughts on the initial demo. I am gonna to build the first page fields, (https://docs.google.com/document/d/1E1THXZ-YhL_Rp5wMKD79O9Hb1ynQsrlqagnNJE3EPr8/edit#heading=h.y0hfixhkxv4x)

And if there is something you think could be better for me to arrange components and something I need to take care of for the next push, feel free to comment, thanks ^^

File: package.json [hunk 1]
```diff
@@ -3,9 +3,11 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@material-ui/core": "^4.11.0",
```
Comment (autumnfound) [inline package.json:6]: If we plan on bundling material-ui and formik, be sure to include them in the NOTICE.md file. As bundling strips comments, we need to disclose explicitly that we are making use of their code!

File: src/components/forms/formStyles.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { makeStyles } from "@material-ui/core/styles";
+
+export const useStyles = makeStyles(theme => ({
+  form: {
+    display: "flex",
```
Comment (autumnfound) [inline src/components/forms/formStyles.js:5]: Be careful around the usage of flex, as anything beyond basic cases can run into issues in IE. The policy on IE support is best efforts, and make sure it doesn't look awful. As long as the code breaks gracefully, we should be good. https://caniuse.com/flexbox

**Comment by flora8984461** [comment]
Temporarily removed material-ui since we already have bootstrap and I don't have the design yet. Not sure how will I make the design. 

Not sure should I use material-ui? 🤔

**Comment by ericpoirier** [comment]
> Temporarily removed material-ui since we already have bootstrap and I don't have the design yet. Not sure how will I make the design.
> 
> Not sure should I use material-ui? 🤔

I don't think you should be using that since we don't use it anywhere else.

Solstice and Bootstrap should be all you need.

**Comment by flora8984461** [comment]
> > Temporarily removed material-ui since we already have bootstrap and I don't have the design yet. Not sure how will I make the design.
> > Not sure should I use material-ui? 🤔
> 
> I don't think you should be using that since we don't use it anywhere else.
> 
> Solstice and Bootstrap should be all you need.

Sounds fair. I was mainly thinking of some components so that I don't need to build from scratch, but I still need to adjust the styling. For now, it is all good, not quite need.

**Comment by chrisguindon** [comment]
@flora8984461 I would like to merge this soon to make it easier for the team to collaborate. Can you prepare the code for review?

I totally understand that this is still a WIP, I am not expecting perfection here.

**Comment by flora8984461** [comment]
> @flora8984461 I would like to merge this soon to make it easier for the team to collaborate. Can you prepare the code for review?
> 
> I totally understand that this is still a WIP, I am not expecting perfection here.

Yeah, you can merge this for now. I will add my improvements later in other PRs.

**Pull request merged**